src/reducers/7.12/reducer.ts

```diff
@@ -1698,23 +1710,63 @@ export const getConnector = (cr: BrokerCR, name: string) => {
   return undefined;
 };
 
+export const getAllConnectors = (brokerModel: BrokerCR) => {
+  return brokerModel.spec?.connectors || [];
+};
+
+export const generatePortNumber = (usedPorts: number[]): number => {
+  let port = 5555;
+  while (usedPorts.includes(port)) {
+    port += 1;
+    //console.log('increament port', port);
+  }
+  return port;
+};
```

Comment:
There's no need to work with a secondary list of `usedPorts` we can work in place instead. To get a new free port:
Find the biggest port number in all the acceptors, connectors and the console. Then just add +1 to it.